specification/schemas/Predicate.json

```diff
@@ -0,0 +1,57 @@
+{
+  "type": "object",
+  "required": [
+    "operator",
+    "left_value",
+    "right_value"
+  ],
+  "additionalProperties": false,
+  "properties": {
+    "operator": {
+      "type": "string",
+      "enum": [
+        "==",
+        "!=",
+        ">",
+        "<",
+        ">=",
+        "<=",
+        "AND",
+        "OR"
+      ],
+      "example": "=="
+    },
+    "left_value": {
+      "oneOf": [
+        {
+          "$ref": "#/components/schemas/PredicateValue"
+        },
+        {
+          "$ref": "#/components/schemas/ScalarValue"
+        },
+        {
+          "$ref": "#/components/schemas/PathValue"
+        },
+        {
+          "$ref": "#/components/schemas/ResourceValue"
+        }
+      ]
```

Comment:
Instead of doing this `oneOf` everywhere, you could create a `Value` schema that holds this `oneOf` and reference that everywhere.